features/schedule/basic_workflow/feature.go

```diff
@@ -0,0 +1,122 @@
+package basic_workflow
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/sdk-features/harness/go/harness"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/workflow"
+)
+
+var Feature = harness.Feature{
+	Workflows: BasicScheduleWorkflow,
+	Execute:   Execute,
+}
+
+func BasicScheduleWorkflow(ctx workflow.Context, arg string) (string, error) {
+	return arg, nil
+}
+
+func Execute(ctx context.Context, r *harness.Runner) (client.WorkflowRun, error) {
+	// Start schedule every 2s
+	workflowID := uuid.NewString()
+	handle, err := r.Client.ScheduleClient().Create(ctx, client.ScheduleOptions{
+		// TODO(cretz): Required? https://github.com/temporalio/sdk-go/issues/957
+		ID:      uuid.NewString(),
+		Spec:    client.ScheduleSpec{Intervals: []client.ScheduleIntervalSpec{{Every: 2 * time.Second}}},
+		Overlap: enums.SCHEDULE_OVERLAP_POLICY_BUFFER_ONE,
+		// TODO(cretz): Would prefer if this were a pointer. See
+		// https://github.com/temporalio/sdk-go/issues/958.
+		Action: client.ScheduleWorkflowAction{
+			ID:        workflowID,
+			Workflow:  BasicScheduleWorkflow,
+			Args:      []interface{}{"arg1"},
+			TaskQueue: r.TaskQueue,
+		},
+	})
+	r.Require.NoError(err)
+
+	// Remove the schedule on complete
+	defer func() {
+		if err := handle.Delete(context.Background()); err != nil {
+			r.Log.Warn("Failed deleting schedule handle", "error", err)
+		}
+	}()
+
+	// Confirm simple describe
+	desc, err := handle.Describe(ctx)
+	r.Require.NoError(err)
+	r.Require.Equal(workflowID, desc.Schedule.Action.(client.ScheduleWorkflowAction).ID)
+
+	// Confirm simple list
+	iter, err := r.Client.ScheduleClient().List(ctx, client.ScheduleListOptions{})
+	r.Require.NoError(err)
+	foundSchedule := false
+	for iter.HasNext() && !foundSchedule {
+		e, err := iter.Next()
+		r.Require.NoError(err)
+		foundSchedule = e.ID == handle.GetID()
+	}
+	r.Require.True(foundSchedule)
+
+	// Wait for first completion
+	waitCompletedWith(ctx, r, workflowID, "arg1")
+
+	// Update and change arg
+	err = handle.Update(ctx, client.ScheduleUpdateOptions{
+		DoUpdate: func(in client.ScheduleUpdateInput) (*client.ScheduleUpdate, error) {
+			update := &client.ScheduleUpdate{Schedule: &in.Description.Schedule}
+			action := update.Schedule.Action.(client.ScheduleWorkflowAction)
+			action.Args = []interface{}{"arg2"}
+			update.Schedule.Action = action
+			return update, nil
+		},
+	})
+	r.Require.NoError(err)
+
+	// Wait for next completion
+	waitCompletedWith(ctx, r, workflowID, "arg2")
+	return nil, nil
+}
+
+func waitCompletedWith(ctx context.Context, r *harness.Runner, id string, untilResult string) {
+	// Wait a max of 10s, waiting 1s between non-error tries
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+	for {
+		// Get all completed
+		req := &workflowservice.ListWorkflowExecutionsRequest{
```

Comment:
I would use `DescribeScedule.info.recentActions` instead of listing all workflow executions. Would be much more easier and that would test a little bit more the schedule implementation itself.

Reply:
I can add that. I still want to list workflows because I want to test the non-schedule result of the schedule on the actual system too.